table/column.go

```diff
@@ -181,7 +181,7 @@ func CastValue(ctx sessionctx.Context, val types.Datum, col *model.ColumnInfo, r
 	if types.ErrOverflow.Equal(err) && returnOverflow {
 		return casted, err
 	}
-	if types.ErrTruncated.Equal(err) {
+	if err != nil && types.ErrTruncated.Equal(err) && col.Tp != mysql.TypeSet && col.Tp != mysql.TypeEnum {
```

Comment:
We don't need `err!=nil` because in `types.ErrTruncated.Equal(err)` fix nil problem.

Reply:
@XuHuaiyu